### PR TITLE
HMA-5085: Compose for TabBarView component

### PR DESCRIPTION
**Pull request by ngoulongkam**
# 📝 Description

Github Issue
https://jira.tools.tax.service.gov.uk/browse/HMA-5080
  
- [ ] Updated CHANGELOG
- [ ] Updated README

# 🛠 Testing Notes
- Test Compose's TabBarView component behaves the same as the existing one.

# 📸 Screenshots
HMA-5085

### CODE REVIEW

File: sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/screens/molecules/TabBarViewScreen.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import uk.gov.hmrc.components.compose.molecule.tabbar.TabBarViewStyle
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme
+import uk.gov.hmrc.sample_compose_components.R
+
+@Composable
+fun TabBarViewScreen() {
+    var lightSelectedTabIndex by remember { mutableStateOf(0) }
+    val lightTabItems = listOf("Light", "Style", "Tabs")
+
+    var darkSelectedTabIndex by remember { mutableStateOf(0) }
```
Comment (aliahmedbd) [inline sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/screens/molecules/TabBarViewScreen.kt:41]:
```suggestion
    var darkSelectedTabIndex by remember { mutableIntStateOf(0) }
```

File: sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/screens/molecules/TabBarViewScreen.kt [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    val darkTabItems = listOf("Dark", "Style", "Tabs")
+
+    var longSelectedTabIndex by remember { mutableStateOf(0) }
```
Comment (aliahmedbd) [inline sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/screens/molecules/TabBarViewScreen.kt:44]:
```suggestion
    var longSelectedTabIndex by remember { mutableIntStateOf(0) }
```

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/tabbar/TabBarView.kt [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    tabBarStyle: TabBarViewStyle = TabBarViewStyle.LIGHT,
+) {
+
+    // Getting the screen's width in DP.
+    val configuration = LocalConfiguration.current
+    val screenWidth = configuration.screenWidthDp.dp
+
+    // Getting each of the tab's width in DP.
+    val tabWidths = tabItems.map { tabText ->
+        (tabText.length * HmrcTheme.typography.tabBarText.fontSize.value).dp
```
Comment (aliahmedbd) [inline components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/tabbar/TabBarView.kt:54]: Do we need to consider minimum padding here for individual tab items?

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/tabbar/TabBarView.kt [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                    text = tabText,
+                    style = HmrcTheme.typography.tabBarText,
+                    modifier = Modifier.padding(HmrcTheme.dimensions.hmrcSpacing16),
+                )
+            }
+        }
+    }
+}
+
+enum class TabBarViewStyle(val backgroundColour: Color, val selectedTextColour: Color) {
```
Comment (emmaatkins) [inline components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/tabbar/TabBarView.kt:143]: Should these not be in with the rest of our Styles maybe?

**Review by aliahmedbd — APPROVED** [review]

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/tabbar/TabBarView.kt (hunk 4 above)
Comment (ngoulongkam) [reply components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/tabbar/TabBarView.kt:143]: My thinking behind this is that these `TabBarViewStyle` only really gets used in this `TabBarView`. This really shouldn't be use anywhere else, hence why I've put it in this class directly. 

Is that ok, Or do you think this should still go somewhere else?

**Review by emmaatkins — APPROVED** [review]

**Pull request merged**